File: DungeonDebt/Assets/Scripts/Data/ShopOffer.cs
public class ShopOffer
{
    public ShopOffer(HeroDefinition hero, int hireCost)
        : this(hero, hireCost, HeroTier.Bronze)
    {
    }

    public ShopOffer(HeroDefinition hero, int hireCost, HeroTier tier)
    {
        Hero = hero;
        HireCost = hireCost;
        Tier = tier;
        Purchased = false;
    }

    public HeroDefinition Hero { get; set; }
    public int HireCost { get; set; }
    public HeroTier Tier { get; set; }
    public bool Purchased { get; set; }
}

File: web/src/data/ShopOffer.js
// Ported from DungeonDebt/Assets/Scripts/Data/ShopOffer.cs
import { HeroTier } from "./enums.js";

export class ShopOffer {
  constructor(hero, hireCost, tier = HeroTier.Bronze) {
    this.hero = hero;
    this.hireCost = hireCost;
    this.tier = tier;
    this.purchased = false;
  }
}
